security-analysis/security-analysis-api/src/test/java/com/powsybl/security/detectors/DefaultLimitViolationDetectorTest.java

```diff
@@ -56,6 +59,31 @@ void detectPermanentLimitOverloadOnSide2OfLine1() {
                     assertEquals(1101, l.getValue(), 0d);
                     assertSame(Branch.Side.TWO, l.getSide());
                     assertEquals(600, l.getAcceptableDuration());
+                    assertEquals(PERMANENT_LIMIT_NAME, l.getLimitName());
+                });
+    }
+
+    @Test
+    void testLimitReductionOnCurrentPermanentLimit() {
+        final double i = 460;
+        Optional<? extends LoadingLimits> line1Limits = line1.getLimits(LimitType.CURRENT, Branch.Side.ONE);
+        assertTrue(line1Limits.isPresent()
+            && line1Limits.get().getTemporaryLimits().isEmpty()
+            && line1Limits.get().getPermanentLimit() > i); // no overload expected
+
+        // no violation if limitReduction is 1
+        DefaultLimitViolationDetector cdetector = new DefaultLimitViolationDetector(1.0f, EnumSet.allOf(LoadingLimitType.class));
+        cdetector.checkLimitViolation(line1, Branch.Side.ONE, i, violationsCollector::add, LimitType.CURRENT);
+        assertTrue(violationsCollector.isEmpty());
+
+        // violation reported if limitReduction is 0.9
+        cdetector = new DefaultLimitViolationDetector(0.9f, EnumSet.allOf(LoadingLimitType.class));
+        cdetector.checkLimitViolation(line1, Branch.Side.ONE, i, violationsCollector::add, LimitType.CURRENT);
+        Assertions.assertThat(violationsCollector)
+                .hasSize(1)
+                .allSatisfy(l -> {
+                    assertEquals(PERMANENT_LIMIT_NAME, l.getLimitName());
+                    assertEquals(0.9f, l.getLimitReduction());
```

Comment:
maybe you can check the limit and the value here to check if it is what is expected

Reply:
done